napari/_qt/utils.py

```diff
@@ -235,16 +236,29 @@ def combine_widgets(
         return widgets.native  # type: ignore
     elif isinstance(widgets, QWidget):
         return widgets
-    elif is_sequence(widgets) and all(isinstance(i, QWidget) for i in widgets):
-        container = QWidget()
-        container.setLayout(QVBoxLayout() if vertical else QHBoxLayout())
-        for widget in widgets:
-            container.layout().addWidget(widget)
-        return container
-    else:
-        raise TypeError(
-            trans._('"widget" must be a QWidget or a sequence of QWidgets')
-        )
+    elif is_sequence(widgets):
+        # the same as above, compatibility with magicgui v0.2.0
+        widgets = [
+            i.native if isinstance(getattr(i, 'native', None), QWidget) else i
+            for i in widgets
+        ]
+        if all(isinstance(i, QWidget) for i in widgets):
+            container = QWidget()
+            container.layout = QVBoxLayout() if vertical else QHBoxLayout()
+            container.setLayout(container.layout)
+            for widget in widgets:
+                container.layout.addWidget(widget)
+            # if this is a vertical layout, and none of the widgets declare a size
```

Comment:
slightly concerned that we're doing this again here (we also do it in dockwidget.__init__) ... did you find that the existing logic wasn't catching this case?

Reply:
This was a mistake on my part. I was using an outdated version of napari and ended up copying some old code.